Add config file

gabrielbdornas opened the pull request:
@Andrelamor, gostaria de mais um teste seu antes de mergiar esta nova funcionalidade.

Andrelamor commented:
@gabrielbdornas 
- o primeiro teste que rodei não funfou pq não tinha editado o `config.py`, então vou incluí-lo no README - vide [outro PR]();
- o segundo teste funcionou perfeitamente, como eu te printei no teams, ressalvas a seguir:
- os nomes das abas do arquivo em excel vieram para o nome dos arquivos csv criados, então editei o scrip para isso não ocorrer (se tiver uma situação em que vc pensou que isso será necessário, por favor me lembre, pois o [caso concreto do TPCG](https://github.com/transparencia-mg/termos-parceria-contratos-gestao/tree/main/upload) é aproveitar outra planilha com um nome que difere do repo/dataset; amanhã também falo com o Gustavo para ele modificar os nomes das abas pra bater com o [schema atual](https://github.com/transparencia-mg/termos-parceria-contratos-gestao/blob/main/dataset/datapackage.yaml);
- aproveitei este PR pra mudar a versão do dpckan para 22 (se isso estiver ok no fluxo, descarto o #37 no qual queria selecionar apenas o [commit específico que muda o requirements.txt](https://github.com/transparencia-mg/new-dataset-template/pull/37/commits/2a0708136e934a17901de40fc228f59bdd855f7d);
-

Andrelamor commented:
@gabrielbdornas segue o erro:

````
env)
m6695506@D03GC5234264738 MINGW64 ~/Documents/CGE_transparência/repositorios/transparencia-mg/new-dataset-template (add-config-file)
$ task csv
Traceback (most recent call last):
  File "C:\Users\m6695506\Documents\CGE_transparência\repositorios\transparencia
-mg\new-dataset-template\scripts\convert_csv.py", line 59, in <module>
    find_all_files()
  File "C:\Users\m6695506\Documents\CGE_transparência\repositorios\transparencia
-mg\new-dataset-template\scripts\convert_csv.py", line 16, in find_all_files
    convert_csv(file)
  File "C:\Users\m6695506\Documents\CGE_transparência\repositorios\transparencia
-mg\new-dataset-template\scripts\convert_csv.py", line 27, in convert_csv
    new_columns.append(snake_small_case(column))
  File "C:\Users\m6695506\Documents\CGE_transparência\repositorios\transparencia
-mg\new-dataset-template\scripts\convert_csv.py", line 50, in snake_small_case
    column_lower = column.lower()
AttributeError: 'datetime.datetime' object has no attribute 'lower'
````

gabrielbdornas commented:
> @gabrielbdornas segue o erro:
> 
> ```
> env)
> m6695506@D03GC5234264738 MINGW64 ~/Documents/CGE_transparência/repositorios/transparencia-mg/new-dataset-template (add-config-file)
> $ task csv
> Traceback (most recent call last):
>   File "C:\Users\m6695506\Documents\CGE_transparência\repositorios\transparencia
> -mg\new-dataset-template\scripts\convert_csv.py", line 59, in <module>
>     find_all_files()
>   File "C:\Users\m6695506\Documents\CGE_transparência\repositorios\transparencia
> -mg\new-dataset-template\scripts\convert_csv.py", line 16, in find_all_files
>     convert_csv(file)
>   File "C:\Users\m6695506\Documents\CGE_transparência\repositorios\transparencia
> -mg\new-dataset-template\scripts\convert_csv.py", line 27, in convert_csv
>     new_columns.append(snake_small_case(column))
>   File "C:\Users\m6695506\Documents\CGE_transparência\repositorios\transparencia
> -mg\new-dataset-template\scripts\convert_csv.py", line 50, in snake_small_case
>     column_lower = column.lower()
> AttributeError: 'datetime.datetime' object has no attribute 'lower'
> ```

@Andrelamor, não consegui reproduzir o erro relatado acima. Para tentar reproduzir o mesmo executei os seguintes passos:

- Clonei o repositório [termos-parceria-contratos-gestao](https://github.com/transparencia-mg/termos-parceria-contratos-gestao) localmente.
- Criei, ativei e instalei os pacotes Python.
- Adicionei o remote `upstream` apontando para o repositório forkado [new-dataset-template](https://github.com/transparencia-mg/new-dataset-template) com o comando `git remote add upstream git@github.com:transparencia-mg/new-dataset-template.git`.
- Busquei a branch `upstream/add-config-file` com o comando `git fetch upstream`.
- Criei a branch `test-config-file` com o comando `git checkout -b test-config-file`.
- Mergiei a branch `upstream/add-config-file` a esta nova branch com o comando `git merge upstream/add-config-file`.
- Corrigi os conflitos no arquivo `convert_csv.py`.
- Rodei o comando `task csv` e os arquivos foram gerados normalmente, sem nenhum erro.

```
(venv) ➜  termos-parceria-contratos-gestao git:(test-config-file) task csv
(venv) ➜  termos-parceria-contratos-gestao git:(test-config-file) ✗ gst                               
On branch test-config-file
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   dataset/data/aditivos.csv
	modified:   dataset/data/relatorios.csv
	modified:   dataset/data/repasses.csv
	modified:   dataset/data/termo-parceria-contrato-gestao.csv

no changes added to commit (use "git add" and/or "git commit -a")
```

Andrelamor commented:
@gabrielbdornas a msg de erro aponta para linhas do script, de snake lower case; que conflitos apareceram?

Andrelamor commented:
tb fiz um merg no repo de termos de parceria para concluir o teste via GUI: https://github.com/transparencia-mg/termos-parceria-contratos-gestao/commit/42aadbf0eafa97a1f64d97be04c1edcd02d9d029
